xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-livetable/src/main/java/org/xwiki/livedata/internal/livetable/ModelBridge.java

```diff
@@ -0,0 +1,294 @@
+    }
+
+    /**
+     * Update all the provided properties of the document. The properties can either be from the document itself, if
+     * prefixed with {@code doc.}, or from an XObject instance of the specified type (class reference). The XObject at
+     * the provided index number is updated.
+     *
+     * @param entry the entry to update
+     * @param documentReference the document to update
+     * @param classReference the type of XObject to update
+     * @param objectNumber the index of the XObject to update
+     * @throws AccessDeniedException in case the current user is not allow to edit the document
+     * @throws XWikiException in case of error when loading or saving the updated document
+     * @throws LiveDataException in case of error when validating the document
+     * @see #updateAll(Map, DocumentReference, DocumentReference)
+     */
+    public void updateAll(Map<String, Object> entry, DocumentReference documentReference,
+        DocumentReference classReference, int objectNumber)
+        throws AccessDeniedException, XWikiException, LiveDataException
+    {
+        this.authorization.checkAccess(Right.EDIT, documentReference);
+        XWikiContext xcontext = this.xcontextProvider.get();
+        XWikiDocument document = xcontext.getWiki().getDocument(documentReference, xcontext);
+
+        if (document.isNew()) {
+            throw new LiveDataException(NEW_DOCUMENT_UPDATE_ERROR);
+        }
+
+        convertPropertiesFromHtml(entry, this.localSerializer.serialize(classReference), objectNumber);
+
+        for (Map.Entry<String, Object> property : entry.entrySet()) {
+            this.updateProperty(property.getKey(), property.getValue(), classReference, objectNumber, document);
+        }
+
+        saveDocument(document);
+    }
+
+    private void convertPropertiesFromHtml(Map<String, Object> entry, String className, int objectNumber)
+    {
+        if (entry.containsKey(REQUIRES_HTML_CONVERSION)) {
+            String requiresHTMLConversion = (String) entry.get(REQUIRES_HTML_CONVERSION);
+            entry.remove(REQUIRES_HTML_CONVERSION);
+            for (String requiresHTMLConversionProperty : requiresHTMLConversion.split(",")) {
+                // Removes the class name and object index from the conversion property.
+                String propertyName = requiresHTMLConversionProperty.replace(className, "")
+                    .replaceFirst(String.format("_%d_", objectNumber), "");
+                String syntaxKey = propertyName + "_syntax";
+                String cacheKey = propertyName + "_cache";
+                if (entry.containsKey(propertyName)) {
+                    String property = (String) entry.get(propertyName);
+                    entry.put(propertyName, this.htmlConverter.fromHTML(property, (String) entry.get(syntaxKey)));
+                }
+                entry.remove(syntaxKey);
+                entry.remove(cacheKey);
+            }
+        }
+    }
```

Comment:
This PR is already large. I propose to improve this code to avoid code duplication on another commit.